citizennet/purescript-ocelot
# AS-1452 Add MultiInput variant of Typeahead

## What does this pull request do?

Add a new variant of `Typeahead` (`asyncMultiInput` and `syncMultiInput`) that combines `MultiInput` text field and `Typeahead` fuzzy search dropdown for Tags field.

Talked to Dave on potential solutions. Ideally we want to rebuild `MultiInput` using `Select` logic within `Typeahead`. As an intermediate step, we nest current `MultiInput` inside `Typeahead` and synchronize state between using `Query` and `Output` (both manage a list of selections which has to be in sync).

## How should this be manually tested?

- [x] `make` and then open `dist/index.html#typeaheads`
- [x] go to `Typeaheads - Multi-Input` section
- [x] click on the tag field under `Standard`
  - typeahead dropdown should show up
- [x] type in something and select an option using mouse click
  - typeahead fuzzy search should narrow available options
  - selection should be committed to the list in MultiInput
  - open typeahead dropdown and the selected option should be filtered out
- [x] type in something, navigate between options using up/down arrow keys, select an option by pressing `Enter`
  - selection should be committed to the list in MultiInput
  - open typeahead dropdown and the selected option should be filtered out
- [x] remove a selected item by clicking `x` button
  - open typeahead dropdown and the removed option should be available again
- [x] click on a selected item, edit the result, and then press `Enter`
  - change should be committed
  - open typeahead dropdown and the previous selection should be available again
- [x] type in something that's not initially available in the typeahead dropdown and press `Enter`
  - the new keyword should be commited to the list in MultiInput
